scripts/apps/archive/directives/MarkedItemTitle.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * @module superdesk.apps.archive
+ * @ngdoc directive
+ * @name sdMarkedItemTitle
+ * @requires desks
+ * @requires authoring
+ * @requires highlightServce
+ * @requires $timeout
+ * @description
+ *   This directive is used in authoring-topbar to mark a story as a highlight or
+ *   to mark it for a desk.
+ */
+MarkedItemTitle.$inject = ['desks', '$timeout', 'authoring', 'highlightsService'];
+export function MarkedItemTitle(desks, $timeout, authoring, highlightsService) {
+    return {
+        scope: {
+            item: '=item',
+            markField: '@field'
+        },
+        templateUrl: 'scripts/apps/archive/views/marked_item_title.html',
+        // todo(petr): refactor to use popover-template once angular-bootstrap 0.13 is out
+        link: function(scope, elem) {
+            /*
+             * Toggle 'open' class on dropdown menu element
+             * @param {string} isOpen
+             */
+            scope.toggleClass = function(isOpen) {
+                scope.open = isOpen;
+            };
+
+            scope.marks = scope.item[scope.markField];
+            scope.hasMarkItemPrivilege = authoring.itemActions(scope.item).mark_item_for_desks;
+
+            scope.$on('item:' + scope.markField, ($event, data) => {
+                let marks = scope.marks || [];
+                let mark = data.mark_id;
+
+                if (scope.markField === 'marked_desks') {
+                    mark = {desk_id: data.mark_id};
+                }
+
+                if (scope.item._id === data.item_id) {
+                    scope.$apply(() => {
+                        if (data.marked) {
+                            scope.marks = marks.concat(mark);
+                        } else {
+                            _.remove(marks, _.filter(marks, mark)[0]);
+                            scope.marks = marks;
+                        }
+                    });
+                }
+            });
+
+            // Check if the directive is created to mark desks or highlights
+            if (scope.markField === 'marked_desks') {
```

Comment:
You have already written this `if` statement above. Why not include the body of it there? We don't have to have the same `if` statement twice in the same block. Could we merge instead?